src/frontend/src/instance/jaeger.rs

```diff
@@ -0,0 +1,330 @@
+            filters.push(col(TIMESTAMP_COLUMN).gt_eq(lit_timestamp_nano(start_time)));
+        }
+
+        if let Some(end_time) = query_params.end_time {
+            filters.push(col(TIMESTAMP_COLUMN).lt_eq(lit_timestamp_nano(end_time)));
+        }
+
+        if let Some(min_duration) = query_params.min_duration {
+            filters.push(col(DURATION_NANO_COLUMN).gt_eq(lit(min_duration)));
+        }
+
+        if let Some(max_duration) = query_params.max_duration {
+            filters.push(col(DURATION_NANO_COLUMN).lt_eq(lit(max_duration)));
+        }
+
+        Ok(query_trace_table(
+            ctx,
+            self.catalog_manager(),
+            self.query_engine(),
+            selects,
+            filters,
+            Some(DEFAULT_LIMIT),
+            query_params.tags,
+            false,
+        )
+        .await?)
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn query_trace_table(
+    ctx: QueryContextRef,
+    catalog_manager: &CatalogManagerRef,
+    query_engine: &QueryEngineRef,
+    selects: Vec<Expr>,
+    filters: Vec<Expr>,
+    limit: Option<usize>,
+    tags: Option<HashMap<String, JsonValue>>,
+    distinct: bool,
+) -> ServerResult<Output> {
+    let db = ctx.get_db_string();
+    let table = catalog_manager
+        .table(ctx.current_catalog(), &db, TRACE_TABLE_NAME, Some(&ctx))
+        .await
+        .context(CatalogSnafu)?
+        .with_context(|| TableNotFoundSnafu {
+            table: TRACE_TABLE_NAME,
+            catalog: ctx.current_catalog().to_string(),
+            schema: db.to_string(),
+        })?;
+
+    let df_context = create_df_context(query_engine, ctx.clone())?;
+
+    let dataframe = df_context
+        .read_table(Arc::new(DfTableProviderAdapter::new(table)))
+        .context(DataFusionSnafu)?;
+
+    let dataframe = dataframe.select(selects).context(DataFusionSnafu)?;
+
+    // Apply all filters.
+    let dataframe = filters
+        .into_iter()
+        .chain(tags.map_or(Ok(vec![]), |t| tags_filters(&dataframe, t))?)
+        .try_fold(dataframe, |df, expr| {
+            df.filter(expr).context(DataFusionSnafu)
+        })?;
+
+    // Apply the distinct if needed.
+    let dataframe = if distinct {
+        dataframe.distinct().context(DataFusionSnafu)?
+    } else {
+        dataframe
+    };
+
+    // Apply the limit if needed.
+    let dataframe = if let Some(limit) = limit {
+        dataframe.limit(0, Some(limit)).context(DataFusionSnafu)?
+    } else {
+        dataframe
+    };
+
+    // Execute the query and collect the result.
+    let stream = dataframe.execute_stream().await.context(DataFusionSnafu)?;
+
+    let output = Output::new_with_stream(Box::pin(
+        RecordBatchStreamAdapter::try_new(stream).context(CollectRecordbatchSnafu)?,
+    ));
+
+    Ok(output)
+}
+
+// To decouple from the global session context, it will create a new session context and register the necessary UDFs.
+fn create_df_context(
```

Comment:
Any background for this? Is it really needed to use a separated context?

Reply:
Guide by @evenyag . Also, I add the following comment for background:

```
// The current implementation registers UDFs during the planning stage, which makes it difficult
// to utilize them through DataFrame APIs. To address this limitation, we create a new session
// context and register the required UDFs, allowing them to be decoupled from the global context.
```